ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -78,6 +78,61 @@ objects:
 
     containers:
 
+    - name: lease
+      image: ci/boskoscli:latest
+      terminationMessagePolicy: FallbackToLogsOnError
+      resources:
+        requests:
+          cpu: 10m
+          memory: 10Mi
+        limits:
+          memory: 200Mi
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp/shared
+      - name: cluster-profile
+        mountPath: /tmp/cluster
+      - name: artifacts
+        mountPath: /tmp/artifacts
+      env:
+      - name: CLUSTER_TYPE
+        value: ${CLUSTER_TYPE}
+      - name: CLUSTER_NAME
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      command:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        if [[ "${CLUSTER_TYPE:-}" != "aws" ]]; then
+          echo "[WARNING] No resource leasing is defined for \"${CLUSTER_TYPE}\" clusters"
+          exit 0
+        fi
+
+        resource="$( boskosctl --server-url http://boskos --owner-name "${CLUSTER_NAME}" acquire --type aws-quota-slice --state clean --target-state leased --timeout 30m )"
+        touch /tmp/shared/leased
+
+        function release() {
+            local resource_name; resource_name="$( jq .name <<<"${resource}" )"
+            boskosctl --server-url http://boskos --owner-name "${CLUSTER_NAME}" release --name "${resource_name}" --target-state clean
+        }
+        trap release EXIT
+
+        boskosctl --server-url http://boskos --owner-name "${CLUSTER_NAME}" heartbeat --resource "${resource}" --period 30s --timeout 4h &
```

Comment:
This is polling every 30s to extend the lease, so why extend it to 4h?  After `/tmp/shared/exit` is touched, you just need enough time for [the teardown logic][1], so polling every 30s with a 1h extension seems like it would be plenty.  And actually, since you're backgrounding this, you don't need a low polling period for responsiveness, so you could poll every 10m with a 1h extension.

[1]: https://github.com/openshift/release/blob/44dd3116723fe2b4ec7ea4a8317aa48a418fd7b6/ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml#L805-L924

Reply:
`4h` is the time before the job gets nuked by `prow` so I figured we could heartbeat as long as possible

Reply:
> 4h is the time before the job gets nuked by prow so I figured we could heartbeat as long as possible

But that's 4h after the job began, so if you set that to 4h away you wouldn't need to heartbeat, you could just request a one-off 4h lease and exit, right?  Ideally we reclaim those resources as soon as the cluster is cleaned up, not 4h after the `lease` container notices the `exit` file.

If we wanted to cut it close, we'd have the `teardown` container touch a new file when teardown successfully completed, stop heartbeating, and release the lease then.  With that approach, we could use 30s polls with a 2m lease or something else in that short ballpark.

If we don't want to bother with `teardown` -> `lease` container feedback outside of `exit`, we only need the lease to last long enough to cover log-gathering and resource destruction after a failed `setup` or `test` touched `exit`, and that should come in well under 4h since the last heartbeat.